core/cfg.py

```diff
@@ -165,19 +161,26 @@ def __str__(self):
         return "{0.source} -- {0.condition} -- {0.target}".format(self)
 
 
-class ControlFlowGraph(object):
-    def __init__(self, nodes: Set[Node], in_node: Node, out_node: Node, edges: Set[Edge]):
+class ControlFlowGraph:
+    def __init__(self, nodes: Set[Node], in_node: Node, out_node: Node, edges: Set[Edge] = set(), loose_in_edges=None,
+                 loose_out_edges=None):
         """Control flow graph representation.
         
-        :param nodes: nodes of the control flow graph
+        :param nodes: set of nodes of the control flow graph
         :param in_node: entry node of the control flow graph
         :param out_node: exit node of the control flow graph
-        :param edges: edges of the control flow graph
+        :param edges: optional set of edges of the control flow graph
+        :param loose_in_edges: optional set of loose edges that have no start yet and end inside this CFG
+        :param loose_out_edges: optional set of loose edges that start inside this CFG and have no end yet
```

Comment:
What are these things? Did you add them just for the CFG generation? Can we have the stuff related to the CFG generation in the CFG generator instead?